### DISCUSSION TITLE
Import from firefox

### DISCUSSION THREAD

**Discussion by pizzav-xyz** [Floorp-Projects/Floorp#729] [Q&A]
how to import all data bookmarks and extensions from firefox to floorp

**Comment by regularentropy** [comment]
You need to click “Import data” during your browser setup

**Reply by pizzav-xyz** [reply]
firefox is not an option

**Comment by regularentropy** [comment]
> firefox is not an option

https://support.mozilla.org/en-US/kb/export-firefox-bookmarks-to-backup-or-transfer

**Comment by Telmesomething** [comment]
For the extensions part of your question proceed as follows.

1. in Firefox, type **about:profiles** in the URL bar and hit Enter.
2. In the Root Directory of the Firefox folder which contains the extensions you want to import, click Open Folder.
3. Right click the **extensions** folder and choose Copy. Place it anywhere you want on your machine.
4. Open Floorp and open the extensions folder you just copied. 
5. Next, drag and drop each extension you want to import into Floorp and confirm you want to proceed when the prompt appears. 

Step 5 is the boring part because you have to drag & drop them individually, but at least you don't need a web connection and you don't need to look for them on the FF addons site. 

EDIT: I should perhaps add that the extensions are the alpha-numerical files in the extensions folder each of which has an XPI extension. If you can't see the XPI extension, you have to remove the checkmark from "Hide file extensions for known file types" in File Explorer for whichever version of Windows you have installed.  If you don't know how to do that let me know and I'll be happy to explain the procedure. ;)

**Comment by Telmesomething** [comment]
Something else which should make it easier for you to identify your current extensions and which ones you consider more important than others to install. 

1. Open Firefox and in the URL bar at the top type: **about:support** and hit Enter.
2. Scroll down to the 4th set of data called **Add-ons**. 

You can see all the extensions you have installed in a grid which includes version numbers, whether they're enabled or not and ID data. So when you come across one of those alphanumerical file names which don't provide a clue as to which extension it applies to you can glance at the left hand column and see which one it is. Here's a pix of one taken from an old Firefox I had installed at the time.
![EXTENSIONS-2](https://github.com/Floorp-Projects/Floorp/assets/33827636/1afc2675-55db-4f88-92e2-f666243d5790)

**Comment by Telmesomething** [comment]
Last but not least, an easy way to import your Firefox bookmarks into Floorp. 

1. Hit Ctrl + Shift + O to open the Firefox library.
2. Click **Bookmarks** in the left hand column. This is just to make sure they are all present.
3. Assuming you can see them in the right hand panel click **Import and Backup** at the top and choose **Export Bookmarks to HTML**
4. The file gets saved as **Bookmarks.html**. Place it somewhere where you can find it again easily such as the Desktop for example. 
5. Next step is to open Floorp and type: **about:profiles** in the URL bar and hit Enter. 
6. Find the profile you want to save your Firefox bookmarks to (_if you've only got one it will be called Default_) and click **Open Folder** in the **Root Directory**
7. Scroll down to the file called **places.sqlite** and delete it.
8. Finally, grab the Bookmarks.html file you created earlier and drag and drop it into Floorp.

It sometimes takes a few seconds to populate the bookmarks directory depending on how many there are, but that's essentially all there is to it. The places.sqlite file you deleted earlier will be recreated by Floorp. 

If you've been using Floorp for a while and have created a few bookmarks already they will be lost. If you want to keep them, create a new Bookmarks.html file in the same way you just did for Firefox and save it somewhere. Then create a new Profile and follow the same procedure to import them. You can open that new profile via a link in the Settings menu which will open it in a new window.
